includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -10,34 +10,30 @@
  *
  * Extending main gateway class and only loaded if Subscriptions is available.
  */
-class WC_Gateway_Amazon_Payments_Advanced_Subscriptions extends WC_Gateway_Amazon_Payments_Advanced {
+class WC_Gateway_Amazon_Payments_Advanced_Subscriptions {
```

Comment:
the extend of WC_Gateway... is not needed for this to work, to be loaded by woocommerce?

Reply:
I've refactored the class and implemented filters so that it's not needed. Otherwise it will be inheritance hell with legacy and non-legacy implementations on different subclasses, some things in common, with legacy and non-legacy code split up in separate classes.

It's typically implemented as inheriting the base class, but i thought it might be cleaner to have it like this. Let me know if you see any specific problem with this.

Before moving on to QA I'll do a pass myself on v1 functionality not breaking because of this further refactor.